testtools/_deferred.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) testtools developers. See LICENSE for details.
+
+"""Utilities for Deferreds."""
+
+from functools import partial
+
+from testtools.content import TracebackContent
+
+
+class DeferredNotFired(Exception):
+    """Raised when we extract a result from a Deferred that's not fired yet."""
+
+    def __init__(self, deferred):
+        msg = "%r has not fired yet." % (deferred,)
+        super(DeferredNotFired, self).__init__(msg)
```

Comment:
The main thing that I think it would help is testing the implementation... The assertion about an attribute referencing the right object is more direct than an assertion about a string containing the right bytes.  That's not a very compelling user-facing scenario - but sometimes when improving things so they're easier to test, unanticipated improvements are yielded for end-users as well.

Definitely no really strong feelings here, just a vague hunch keeping things structured when possible is better than tossing strings around.